mso/resource_mso_schema_site_service_graph.go

```diff
@@ -210,28 +258,48 @@ func resourceMSOSchemaSiteServiceGraphDelete(d *schema.ResourceData, m interface
 	return nil
 }
 
-func getServiceNodeList(siteServiceNodes interface{}, graphCont *container.Container) []interface{} {
+func getServiceNodeList(msoClient *client.Client, siteServiceNodes interface{}, graphCont *container.Container) ([]interface{}, error) {
 	siteServiceNodeList := make([]interface{}, 0, 1)
-	for index, templateServiceNode := range graphCont.S("serviceNodes").Data().([]interface{}) {
+	for index, serviceNode := range graphCont.S("serviceNodes").Data().([]interface{}) {
+		nodeType, err := getNodeNameFromId(msoClient, serviceNode.(map[string]interface{})["serviceNodeTypeId"].(string))
+		if err != nil {
+			return nil, err
+		}
+
 		siteServiceNodeMap := siteServiceNodes.([]interface{})[index].(map[string]interface{})
+
+		provider_connector_type_value := siteServiceNodeMap["provider_connector_type"]
+		if nodeType == "firewall" {
+			provider_connector_type_value = siteServiceNodeMap["firewall_provider_connector_type"]
+		}
+
 		serviceNodeMap := map[string]interface{}{
-			"serviceNodeRef": templateServiceNode.(map[string]interface{})["serviceNodeRef"],
+			"serviceNodeRef": serviceNode.(map[string]interface{})["serviceNodeRef"],
 			"device": map[string]interface{}{
 				"dn": siteServiceNodeMap["device_dn"],
 			},
+			"consumerConnectorType": siteServiceNodeMap["consumer_connector_type"],
+			"providerConnectorType": provider_connector_type_value,
+			"consumerInterface":     siteServiceNodeMap["consumer_interface"],
+			"providerInterface":     siteServiceNodeMap["provider_interface"],
 		}
 		siteServiceNodeList = append(siteServiceNodeList, serviceNodeMap)
 	}
-	return siteServiceNodeList
+	return siteServiceNodeList, nil
 }
 
 func setServiceNodeList(graphCont *container.Container) ([]interface{}, error) {
 	serviceNodeList := make([]interface{}, 0, 1)
 	for _, val := range graphCont.S("serviceNodes").Data().([]interface{}) {
-		device := val.(map[string]interface{})["device"].(map[string]interface{})["dn"]
 		serviceNodeMap := map[string]interface{}{
-			"device_dn": device,
+			"device_dn":                        val.(map[string]interface{})["device"].(map[string]interface{})["dn"],
+			"consumer_connector_type":          val.(map[string]interface{})["consumerConnectorType"],
+			"provider_connector_type":          val.(map[string]interface{})["providerConnectorType"],
+			"consumer_interface":               val.(map[string]interface{})["consumerInterface"],
+			"provider_interface":               val.(map[string]interface{})["providerInterface"],
+			"firewall_provider_connector_type": val.(map[string]interface{})["providerConnectorType"],
```

Comment:
Is this correct that we set same value for both firewall and normal provider connector type? Or should this be one or the other being set?